awsx/ec2/vpc.ts

```diff
@@ -85,67 +88,12 @@ export class Vpc extends schema.Vpc<VpcData> {
     const allocationIds = args.natGateways?.elasticIpAllocationIds ?? [];
     validateEips(natGatewayStrategy, allocationIds, availabilityZones);
 
-    const cidrBlock = args.cidrBlock ?? "10.0.0.0/16";
-
-    const parsedSpecs = validateAndNormalizeSubnetInputs(
-      args.subnetSpecs,
-      availabilityZones.length,
-    );
-
     const subnetStrategy = args.subnetStrategy ?? "Legacy";
-    const subnetSpecs = (() => {
-      if (parsedSpecs === undefined || subnetStrategy === "Legacy") {
-        const legacySubnetSpecs = getSubnetSpecsLegacy(
-          name,
-          cidrBlock,
-          availabilityZones,
-          parsedSpecs?.normalizedSpecs,
-        );
-        return legacySubnetSpecs;
-      }
-
-      if (parsedSpecs.isExplicitLayout) {
-        return getSubnetSpecsExplicit(name, availabilityZones, parsedSpecs.normalizedSpecs);
-      }
-      return getSubnetSpecs(
-        name,
-        cidrBlock,
-        availabilityZones,
-        parsedSpecs.normalizedSpecs,
-        args.availabilityZoneCidrMask,
-      );
-    })();
-
-    let subnetLayout = parsedSpecs?.normalizedSpecs;
-    if (subnetStrategy === "Legacy" || subnetLayout === undefined) {
-      subnetLayout = extractSubnetSpecInputFromLegacyLayout(subnetSpecs, name, availabilityZones);
-    }
-    // Only warn if they're using a custom, non-explicit layout and haven't specified a strategy.
-    if (
-      args.subnetStrategy === undefined &&
-      parsedSpecs !== undefined &&
-      parsedSpecs.isExplicitLayout === false
-    ) {
-      pulumi.log.warn(
-        `The default subnetStrategy will change from "Legacy" to "Auto" in the next major version. Please specify the subnetStrategy explicitly. The current subnet layout can be specified via "Auto" as:\n\n${JSON.stringify(
-          subnetLayout,
-          undefined,
-          2,
-        )}`,
-        this,
-      );
-    }
-
-    validateSubnets(subnetSpecs, getOverlappingSubnets);
-
-    if (subnetStrategy === "Exact") {
-      validateNoGaps(cidrBlock, subnetSpecs);
-    }
-
-    validateNatGatewayStrategy(natGatewayStrategy, subnetSpecs);
 
     const sharedTags = { Name: name, ...args.tags };
 
+    const cidrBlock = Vpc.decideCidrBlockVpcInput(args);
```

Comment:
Should we also validate here that if the user provides the ipamPoolId then they also provide the netmask?

Reply:
This validation is done by the underlying awx.ec2.Vpc resource. It probably is slightly better UX to lift the validation to the awsx.ec2.Vpc layer. We can do this.